### PR TITLE
Updates `mssql` dependencies

### PR DESCRIPTION
**Pull request by bit-shifter**
- `mssql` from `^8.1.2` - `^10.0.0`, and
 - `@types/mssql` from `^6.0.8` - `^9.1.4`

This update brings `@azure/msal-node` to the v2 line, enabling the use of node version 20 for consuming projects.

Hopefully resolving issue #1617

Executed `./scripts/run help` to generate the `yarn.lock` file

Removes yarn-error.log

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -176,4 +176,4 @@
     "**/node-fetch": "^2.6.7",
     "**/markdown-it": "^12.3.2"
   }
-}
+}
```
Comment (BenBirt) [inline package.json:179]: supernit: please bring back the trailing newline

**Review by BenBirt — CHANGES_REQUESTED** [review]
Thanks for deleting `yarn-error.log`!

File: package.json (hunk 1 above)
Comment (bit-shifter) [reply package.json:179]: will do now

**Comment by bit-shifter** [comment]
Are there any checks that need to be done on my end to ensure that the dependency bumps are sound?

The resultant yarn.lock has been checked to verify `@azure/msal-node` - it is now on the 2.x line (supporting node 20).

**Review by BenBirt — APPROVED** [review]

**Comment by BenBirt** [comment]
If all the tests are passing (and they are) - I'm happy!

**Comment by bit-shifter** [comment]
@BenBirt thank for landing, appreciated!

Quick question, when might a new release be triggered?

**Comment by BenBirt** [comment]
I'll probably start one after I merge https://github.com/dataform-co/dataform/pull/1619.

**Pull request merged**